components/sidebar/channel_create.jsx

```diff
@@ -0,0 +1,181 @@
+        );
+
+        const createDirectMessageTooltip = (
+            <Tooltip
+                id='new-group-tooltip'
+                className='hidden-xs'
+            >
+                <FormattedMessage
+                    id='sidebar.createDirectMessage'
+                    defaultMessage='Create new direct message'
+                />
+            </Tooltip>
+        );
+
+        const createPublicChannelIcon = (
+            <OverlayTrigger
+                trigger={tooltipTriggers}
+                delayShow={500}
+                placement='top'
+                overlay={createChannelTootlip}
+            >
+                <button
+                    id='createPublicChannel'
+                    className='add-channel-btn cursor--pointer style--none'
+                    onClick={createPublicChannel}
+                >
+                    {'+'}
+                </button>
+            </OverlayTrigger>
+        );
+
+        const createPrivateChannelIcon = (
+            <OverlayTrigger
+                trigger={tooltipTriggers}
+                delayShow={500}
+                placement='top'
+                overlay={createGroupTootlip}
+            >
+                <button
+                    id='createPrivateChannel'
+                    className='add-channel-btn cursor--pointer style--none'
+                    onClick={createPrivateChannel}
+                >
+                    {'+'}
+                </button>
+            </OverlayTrigger>
+        );
+
+        const createDirectMessageIcon = (
+            <OverlayTrigger
+                className='hidden-xs'
+                delayShow={500}
+                placement='top'
+                overlay={createDirectMessageTooltip}
+            >
+                <button
+                    className='add-channel-btn cursor--pointer style--none'
+                    onClick={createDirectMessage}
+                >
+                    {'+'}
+                </button>
+            </OverlayTrigger>
+        );
+
+        const createPublicDirectChannelTooltip = (
+            <Tooltip
+                id='new-group-tooltip'
+                className='hidden-xs'
+            >
+                <FormattedMessage
+                    id='sidebar.createPublicPrivateChannel'
+                    defaultMessage='Create new public or private channel'
+                />
+            </Tooltip>
+        );
+
+        const createPublicDirectChannelIcon = (
+            <OverlayTrigger
+                className='hidden-xs'
+                delayShow={500}
+                placement='top'
+                overlay={createPublicDirectChannelTooltip}
+            >
+                <button
+                    className='add-channel-btn cursor--pointer style--none'
+                    onClick={createPublicDirectChannel}
+                >
+                    {'+'}
+                </button>
+            </OverlayTrigger>
+        );
+
+        switch (channelType) {
```

Comment:
I'm not sure I understand this component. It feels like it's 4 different components jammed into one since it works completely differently and cares about different things for each value of `channelType` that it's provided. The name `channelType` is also a bit misleading since it also looks like it can take a sorting method as well which doesn't make sense to me.

Reply:
Since it looks like this is supposed to be because the Sidebar just renders a `CreateChannel` component regardless of what is in the section, could you break this up a bit more cleanly to make it obvious what parts are used when? Like you could have
```javascript
renderPublic = () => {
    if (!this.props.canCreatePublicChannel) {
        return null;
    }

    const tooltip = (
        <Tooltip id='new-channel-tooltip' >
            <FormattedMessage
                id='sidebar.createChannel'
                defaultMessage='Create new public channel'
            />
        </Tooltip>
    );

    return (
        <OverlayTrigger
            trigger={tooltipTriggers}
            delayShow={500}
            placement='top'
            overlay={tooltip}
        >
            <button
                id='createPublicChannel'
                className='add-channel-btn cursor--pointer style--none'
                onClick={this.props.createPublicChannel}
            >
                {'+'}
            </button>
        </OverlayTrigger>
    );
};

render() {
    const {sectionType} = this.props;

    switch (sectionType) {
    case 'public':
        return this.renderPublic();
    case 'private':
        return this.renderPrivate();
    case 'direct':
        return this.renderDirect();
    case 'recent':
    case 'alpha':
        return this.renderCombined();
    default:
        return null;
    }
}
```